### ISSUE TITLE
Python API crashes on Google Colab due to TCMalloc

### ISSUE THREAD

**Issue by mewim** [kuzudb/kuzu#1131]
I have identified the cause of the crashes on Google Colab. The cause is that Google replaces the default system memory allocator with its customized implementation `TCMalloc`(https://github.com/google/tcmalloc) globally in Colab environment. However, Kùzu does not seem to work with `TCMalloc` correctly. 

A demo notebook is created on Colab: https://colab.research.google.com/drive/15nPbuynSblR3SaIQl7wv7XMBuT5VMFMq?usp=sharing

The procedure to confirm this cause is as follows:
1. Install Kùzu Python API with `pip`.
1. Run a simple code snippet to load some CSV and do some match queries with Kùzu Python API for 20 times. Note that some crashes will occur.
1. Remove `TCMalloc` from `Colab` environment.
1. Rerun the same code snippet for 20 times. All of them should succeed.

It is not clear yet which part of our code causes the crashes as the crash can occur in multiple places. However, as per https://stackoverflow.com/questions/34146880/segfault-on-tcmalloc-releasetocentralcache, it is likely due to potential memory corruption in our codebase.

Step to reproduce this issue locally in Ubuntu / Debian:
1. Install `tcmalloc`: `sudo apt install libtcmalloc-minimal4`.
1. Install  Kùzu Python API: `sudo pip3 install kuzu`.
1. Copy the code snippet from the Colab notebook, save it locally as `kuzu-demo.py`.
1. Run the Python script with `TCMalloc`: `env LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4 python3 kuzu-demo.py`(Note that the script is less likely to crash locally, but some crashes can still be observed).

Possible solutions:
1. Static-link the system default `malloc` with all our binaries. This shall override the `TCMalloc`.
1. Run our code with `TCMalloc` and use Valgrind or MemorySanitizer to debug and fix the memory issues.
1. Ask user to remove `TCMalloc` from Colab and reload the notebook. However, reloading means crashing the notebook for one time, which is not very good user experience. 

UPDATE: can confirm that Kùzu also does not work correctly with `jemalloc`, but it does not crash as frequently as `TCMalloc`.

**Comment by anuchak** [comment]
This bug does not seem to occur with our latest build (tested locally).
I ran with pip package uninstalled and making it point to the python-api build directory around 50 times, and there were no failures.

However with pip package installed and tcmalloc in the LD_PRELOAD env path, the script definitely fails at least 15-20 times out of 50 runs.

**Comment by mewim** [comment]
I looked into this issue again today and it seems that Google Colab environment has recently been upgraded to `Ubuntu 20.04.5 LTS x86_64`. I compiled the latest master and ran the test code again for 50 times and it looks that they all finishes without any error. Here is a [link to the notebook](https://colab.research.google.com/drive/1xx7wjuGQHw5ae37kGUNbT6rqS1jMR150?usp=sharing). We can close this issue for now.
